torch/ao/quantization/quantizer/x86_inductor_quantizer.py

```diff
@@ -389,6 +506,104 @@ def _annotate_conv2d(
+        getitem_node = list(node.users)[0]
+        if _is_any_annotated([getitem_node, maxpool_node]):
+            return
+        input_node = maxpool_node.args[0]
+        assert isinstance(input_node, Node)
+        input_qspec_map = {}
+        input_qspec_map[input_node] = get_input_act_qspec(quantization_config)
+        maxpool_node.meta[
+            "quantization_annotation"
+        ] = _X86InductorQuantizationAnnotation(
+            input_qspec_map=input_qspec_map,
+            _annotated=True,
+        )
+        getitem_node.meta[
+            "quantization_annotation"
+        ] = _X86InductorQuantizationAnnotation(
+            _annotated=True,
+            _is_output_of_quantized_pattern=True,
+        )
+
+    def _annotation_propagation_quantizable_pattern(
+        self, node: Node, quantization_config: QuantizationConfig
+    ) -> None:
+        # Propagate annotation to quantizable patterns.
+        if (
+            (node.target in quantizable_ops_pt2e)
+            and (not _is_any_annotated([node]))
+            and (node.op == "call_function")
+        ):
+
+            def is_all_inputs_connected_to_quantized_op(input_nodes):
+                # Ensure all the inputs connect to fusion pattern or quantized node
+                for input_node in input_nodes:
+                    if not _is_quantized_op_pt2e(input_node):
+                        return False
+                return True
+
+            if node.target is torch.ops.aten.max_pool2d_with_indices.default:
+                # Recipe of maxpool2d: check input arg[0] of maxpool2d is quantized or not
+                input_nodes_to_check = [node.all_input_nodes[0]]
+                if not is_all_inputs_connected_to_quantized_op(input_nodes_to_check):
+                    return
+                self._annotate_maxpool2d(node, quantization_config)
+                return
+            else:
+                # TODO <leslie>: Enable recipes for more single quantizable op such as view and relu.
+                pass
+        return
+
+    def _annotate_output_for_int8_in_int8_out_pattern(
+        self, node: Node, quantization_config: QuantizationConfig
+    ) -> None:
+        r"""
+        Check and insert observer at output of node in int8_in_int8_out_ops_pt2e if needed.
+        Recipe refers to https://github.com/intel/intel-extension-for-pytorch/blob/
+        90d19323d96afc53fcc22ba5a7bb3fb07fdd6c1c/intel_extension_for_pytorch/quantization/_utils.py#L495
+        """
+        if (node.target in int8_in_int8_out_ops_pt2e) and (_is_any_annotated([node])):
+            if node.target == torch.ops.aten.max_pool2d_with_indices.default:
+                maxpool_node = node
+                assert len(list(maxpool_node.users)) == 1 and (
+                    list(maxpool_node.users)[0].target == operator.getitem
+                )
+                getitem_node = list(node.users)[0]
+                if not _is_all_annotated([getitem_node, maxpool_node]):
```

Comment:
you probably want to skip if any of the node is annotated? otherwise you will be breaking some existing annotations?

Reply:
Oh, I think here we add this check because we expected `getitem_node` and `maxpool_node` already been annotated in previous step of `_annotation_propagation_quantizable_pattern` which should only annotate the inputs of these 2 nodes. Then, here we will annotate the output of `getitem_node` here.

Reply:
I see, I feel this is error prone, since these might be annotated because of other reasons, I think the more robust way to annotate things would be to have a self contained meaningful pattern and just annotate that pattern by itself.

Reply:
Make sense. We may discuss about it further.